app/src/main/java/tech/ula/ui/SessionListActivity.kt

```diff
@@ -85,6 +85,11 @@ class SessionListActivity : AppCompatActivity() {
         }
     }
 
+    private val FILESYSTEM_EXTRACT_LOGGER = { line: String -> Int
+        this@SessionListActivity.runOnUiThread({text_session_list_progress_update_line_2.text = getString(R.string.progress_setting_up_extract_text,line)})
```

Comment:
Did you try this without `@SessionListActivity`? I'm pretty sure `this` should refer to SessionListActivity anyway due to lazy evaluation.

Reply:
I did not.  I read through and example and it is just something they did and we do it in other places like ...
val downloadManager = DownloadUtility(this@SessionListActivity, archType, distType)

Reply:
Here:
while (downloadList.size != downloadedList.size) {
                    
  this@SessionListActivity.runOnUiThread({text_session_list_progress_update_line_2.text = getString(R.string.progress_downloading_out_of,downloadedList.size,downloadList.size)})
                    delay(500)
                }

I have to do it because this is a CorroutineScope

Reply:
Ah, that is why the other line of code i mentioned needs it too.
ok, this one place that you mentioned, should not need it